components/user/UserProfile.js

```diff
@@ -1,18 +1,27 @@
 import React, { useState } from "react";
 import ReactMarkdown from "react-markdown";
 import { MdQrCode2 } from "react-icons/md";
-import { QRCodeSVG } from "qrcode.react";
+import { QRCodeCanvas } from "qrcode.react";
+import { saveAs } from "file-saver";
 
 import FallbackImage from "../FallbackImage";
 import UserSocial from "./UserSocials";
 import Tag from "../Tag";
 import Link from "../Link";
 import Badge from "../Badge";
+import Button from "../Button";
 
 function UserProfile({ BASE_URL, data }) {
   const [qrShow, setQrShow] = useState(false);
   const fallbackImageSize = 120;
 
+  const downloadQR = () => {
+    const qrCode = document.getElementById("qrcode");
+    qrCode.toBlob(function (blob) {
+      saveAs(blob, "qrcode.png");
+    });
+  };
```

Comment:
Both functions should be consistent and use es6 style

```suggestion
  const downloadQR = () => {
    const qrCode = document.getElementById("qrcode");
    qrCode.toBlob((blob) => saveAs(blob, `linkfree-${data.username}.png`));
  };
```

Reply:
Thanks @eddiejaoude for the feedback , username in the file name will look cool .